Intro_to_Supervised_Learning.md

```diff
@@ -0,0 +1,525 @@
+
+![](https://miro.medium.com/max/2872/1*k2bLmeYIG7z7dCyxADedhQ.png)
+
+As we can see here, the Bi's (Beta i's) are the unknown coefficients from our dataset and going back to what you have learned from your Algebra and Pre-Algebra classes, B0 is the y-intercept of the function and B1 is the slope coefficent of the function and X1 is the actual slope. X1 is also the indepencdent variable in our dataset and Yi is the dependent variable in our dataset. Ei (Epsilon i) is the random error in our function where it measures how far off the datapoints of our function are from the bestline of fit.
+
+So how do we determine the value of the Bi's? We determine the value of the Bi's through using the finding the Residual Sum of Squares (RSS). The Residual Sum of Squares is the sum of the amount of variations each data point has from the true value for the best line of fit. The RSS formula can be expressed as this:
+
+![](https://cdn-images-1.medium.com/fit/t/1600/480/1*a0hsu_ldXOKOq88Re4xYtQ.png)
+
+As you can see by this image that it's a summation of the linear regression function where each Xi is a different feature that's given in your dataset. The minimum RSS is achieved when the values of the true and predicted labels are very close which results in the creation of the Bi's.
+
+To measure the accuracy of our model, we will use the Root Mean Square Error (RMSE) formula which is the square root of the RSS of each data point. The formula can be expressed as this:
+
+![](https://secureservercdn.net/160.153.137.16/70j.58d.myftpupload.com/wp-content/uploads/2019/03/rmse-2.png)
+
+Now that we have gotten over the important mathematical concepts to build the regression model, let's put this knowledge into practice. The first step is to import the built in Diabetes dataset from sckiit-learn.
+
+```python
+#Using Diabetes dataset to predict the median blood pressure level for diabetes patients
+from sklearn.datasets import load_diabetes
+diabetes = load_diabetes()
+```
+
+The next step would be to convert the dataframe into a pandas dataframe to create the dataset into tabular form and will define our column names as the feature names of the built in dataset.
+
+```python
+#Converting dataset to Pandas Dataframe
+import pandas as pd
+diabetes_data = pd.DataFrame(diabetes.data, columns = diabetes.feature_names)
+```
+
+The next step would be to check the shape of the dataframe to see how large it is.
+
+```python
+#Seeing the shape of the dataframe
+diabetes_data.shape
+```
+
+```python
+(442, 10)
+```
+
+We can now see here that there are 442 patients in totatl with 10 specific features.
+
+Let's look at the first 5 rows to see what type of features that we're dealing with.
+
```

Comment:
by default, pandas.head() prints first 5 rows. make this clarification too. because if i wanted to see 10 rows its...head(10),